lib/ocpp/v201/charge_point.cpp

```diff
@@ -3588,24 +3588,52 @@ void ChargePoint::handle_get_installed_certificate_ids_req(Call<GetInstalledCert
     this->send<GetInstalledCertificateIdsResponse>(call_result);
 }
 
+bool ChargePoint::should_reject_certificate_install(InstallCertificateUseEnum cert_type) const {
+    const int security_profile = this->device_model->get_value<int>(ControllerComponentVariables::SecurityProfile);
+
+    if (security_profile > 1) {
+        return false;
+    }
+    switch (cert_type) {
+    case InstallCertificateUseEnum::CSMSRootCertificate:
+        return !this->device_model
+                    ->get_optional_value<bool>(
+                        ControllerComponentVariables::AllowCSMSRootCertificateInstallWhenLowSecurityProfile)
+                    .value_or(true);
+
+    case InstallCertificateUseEnum::ManufacturerRootCertificate:
+        return !this->device_model
+                    ->get_optional_value<bool>(
+                        ControllerComponentVariables::AllowManufacturerRootCertificateInstallWhenLowSecurityProfile)
+                    .value_or(true);
+    default:
+        return false;
+    }
+}
+
 void ChargePoint::handle_install_certificate_req(Call<InstallCertificateRequest> call) {
     EVLOG_debug << "Received InstallCertificateRequest: " << call.msg << "\nwith messageId: " << call.uniqueId;
 
     const auto msg = call.msg;
     InstallCertificateResponse response;
 
-    const auto result = this->evse_security->install_ca_certificate(
-        msg.certificate.get(), ocpp::evse_security_conversions::from_ocpp_v201(msg.certificateType));
-    response.status = ocpp::evse_security_conversions::to_ocpp_v201(result);
-
-    if (response.status == InstallCertificateStatusEnum::Accepted) {
-        const auto& security_event = ocpp::security_events::RECONFIGURATIONOFSECURITYPARAMETERS;
-        std::string tech_info =
-            "Installed certificate: " + conversions::install_certificate_use_enum_to_string(msg.certificateType);
-        this->security_event_notification_req(CiString<50>(security_event), CiString<255>(tech_info), true,
-                                              utils::is_critical(security_event));
+    if (should_reject_certificate_install(msg.certificateType)) {
+        response.status = InstallCertificateStatusEnum::Rejected;
+        response.statusInfo = StatusInfo();
+        response.statusInfo->reasonCode = "LowSecurityProfile";
+        response.statusInfo->additionalInfo = "SecurityProfileTooLowForCertificateInstall";
```

Comment:
This one could also be improved a bit. For example "Installation of certificate is not allowed when security profile is less then 2"

Reply:
made it `CertificateInstallationNotAllowedWhenSecurityProfile1`